### ISSUE TITLE
references a decorator that does not exist

### ISSUE THREAD

**Issue by LioMajor** [pmrowla/pylivemaker#54]
* pylivemaker version: 09-05-2020 latest commit
* Python version: 3.7
* Operating System: Win10

lmlsb insertcsv 00000001.lsb 00000001.csv

```
Traceback (most recent call last):
  File "c:\python\lib\site-packages\pylivemaker-1.0.0.dev0-py3.7.egg\livemaker\lsb\novel.py", line 837, in replace_body
    decorator_counts[wd.decorator] += 1
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\python\Scripts\lmlsb-script.py", line 11, in <module>
    load_entry_point('pylivemaker==1.0.0.dev0', 'console_scripts', 'lmlsb')()
  File "c:\python\lib\site-packages\click-7.0-py3.7.egg\click\core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "c:\python\lib\site-packages\click-7.0-py3.7.egg\click\core.py", line 717, in main
    rv = self.invoke(ctx)
  File "c:\python\lib\site-packages\click-7.0-py3.7.egg\click\core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "c:\python\lib\site-packages\click-7.0-py3.7.egg\click\core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\python\lib\site-packages\click-7.0-py3.7.egg\click\core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "c:\python\lib\site-packages\pylivemaker-1.0.0.dev0-py3.7.egg\livemaker\cli\lmlsb.py", line 1335, in insertcsv
    patched = _patch_csv_text(lsb, lsb_file, csv_data, mode=mode)
  File "c:\python\lib\site-packages\pylivemaker-1.0.0.dev0-py3.7.egg\livemaker\cli\lmlsb.py", line 1292, in _patch_csv_text
    scenario.replace_text_blocks(blocks)
  File "c:\python\lib\site-packages\pylivemaker-1.0.0.dev0-py3.7.egg\livemaker\lsb\novel.py", line 901, in replace_text_blocks
    self.replace_body(new_body)
  File "c:\python\lib\site-packages\pylivemaker-1.0.0.dev0-py3.7.egg\livemaker\lsb\novel.py", line 839, in replace_body
    raise BadLnsError("TWd #{} ({}) references a decorator that does not exist.".format(i, wd))
livemaker.exceptions.BadLnsError: TWd #31 (<VAR NAME="押し倒され確率" unk3="0">) references a decorator that does not exist.
```

[00000001.zip](https://github.com/pmrowla/pylivemaker/files/4603464/00000001.zip)

**Comment by pmrowla** [comment]
@LioMajor can you attach the original untranslated lsb? or a previously translated ver that passes `lmlsb validate`?

**Comment by LioMajor** [comment]
Okay, it took some time...
But i found the culprit and the reason for ' references a decorator that does not exist '.

First thing is, because i wanted to translate the game and had text to change what
i mentioned here https://github.com/pmrowla/pylivemaker/issues/51 i had to use
lmcompiler.py from the old tools.

BUT, this doesn't fix the validation byte > mismatch when lmlsb validate

Doing a swift lmlsb extract and lmlsb batchinsert could fix most of the files
except 10 files because there is a bug in lmcompiler.py to handle this correctly:
```
<TXSPF>どうしますか？<STYLE ID="1">押倒<VAR NAME="押し倒され確率" unk3="0">言責<VAR NAME="言葉責め確率" unk3="0">L<VAR NAME="レベル" unk3="0">敵L<VAR NAME="敵レベル" unk3="0">敵HP<VAR NAME="敵HP" unk3="0">増<VAR NAME="増加する性欲" unk3="0"> <VAR NAME="攻撃力補正" unk3="0"></STYLE>
```
I never noticed because the game still just runs fine.

Even importing the fixed culprit 00000001-00004D92.lns doesn't work anymore > error in first post

In the end, i just hex edited the file to put the original code back, exported 00000001-00004D92.lns,
edited "どうしますか？" to "What should i do?" and inserted the line.

And yes, lmcompiler.py x inputfolder outputfolder > lmcompiler.py c inputfolder outputfolder
just breaks it again without doing any changes...

Long story short, it's a mixup problem because i did need to translate parts that
only lmlsb edit does cover (about ~2000 lines...).

**Comment by pmrowla** [comment]
Closing this for now then. Basically, if an LSB is already in an invalid state, it's kind of expected that pylm will not work properly when trying to edit the invalid LSB.
